.pre-commit-config.yaml

```diff
@@ -30,34 +30,17 @@ repos:
     -   id: insert-license
         files: ^(kolena|tests|examples)/.*\.py$
         args: [--license-filepath, "LICENSE_HEADER"]
--   repo: https://github.com/psf/black
-    rev: 23.1.0
-    hooks:
-        -   id: black
-            args: [--line-length, "120"]
--   repo: https://github.com/PyCQA/flake8
-    rev: 5.0.4
-    hooks:
-        -   id: flake8
-            args: [--extend-ignore, "W503,E203,PAR1,PAR002"]
-            additional_dependencies:
-                - flake8-typing-imports==1.12.0
-                - flake8-quotes==3.3.1
-                - flake8-picky-parentheses==0.4.0
 -   repo: https://github.com/pre-commit/mirrors-mypy
     rev: v0.910
     hooks:
       - id: mypy
         name: mypy-default
-        exclude: ^examples/(dataset|workflow)/
-        additional_dependencies: [ types-all, "pydantic<2.0" ]
-      - id: mypy
-        name: mypy-examples-dataset
-        files: ^examples/dataset/
-        additional_dependencies: [ types-all, "pydantic<2.0" ]
-      - id: mypy
-        name: mypy-examples-workflow
-        files: ^examples/workflow/
+        exclude: |
+            (?x)^(
+                examples/.*|
+                tests/.* |
+                kolena/_experimental/.*
```

Comment:
After we resolve the remaing issues, we can try removing the exclude on `kolena/_experimental/.*` (adds another 100 mypy issues)

Reply:
Can you elaborate on this change? Why are `mypy-examples-XXX` removed?

Reply:
Added `mypy-examples-dataset` back. We'll add `mypy-examples-workflow` back in a follow-up — many typing issues there were unmasked by this PR